### PR TITLE
fix: updating location for capacity constraints

### PR DESCRIPTION
**Pull request by ChrisSidebotham**
## Description

Following PR #700. Capacity in WestEurope is running low for compute. Qouta limits have been imposed on non priority subscriptions ultimately causing the deployment test to fail. Deployment Test has now been changed to North Europe. And the subscription Quota should be checked before merging.

| Pipeline | 
| -------- | 
| [avm.res.web.serverfarm](https://github.com/Azure/bicep-registry-modules/actions/workflows/avm.res.web.serverfarm.yml) |

### CODE REVIEW

File: avm/res/web/serverfarm/main.bicep [hunk 1]
```diff
@@ -110,7 +110,7 @@ resource avmTelemetry 'Microsoft.Resources/deployments@2023-07-01' = if (enableT
 resource appServicePlan 'Microsoft.Web/serverfarms@2022-09-01' = {
   name: name
   kind: kind
-  location: location
+  location: 'eastus'
```
Comment (eriqua) [inline avm/res/web/serverfarm/main.bicep:113]: Is it required to be hardcoded in the template?

Comment (ChrisSidebotham) [reply avm/res/web/serverfarm/main.bicep:113]: After discussion with Alex it is due to the parameter being over written in the pipeline - this current change just needs to be to be a per module change and therefore must be hardcoded to the location value

Comment (ChrisSidebotham) [reply avm/res/web/serverfarm/main.bicep]: Updated! fixed `main.bicep`

File: avm/res/web/serverfarm/tests/e2e/max/dependencies.bicep [hunk 2]
```diff
@@ -6,7 +6,7 @@ param managedIdentityName string
 
 resource managedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2018-11-30' = {
   name: managedIdentityName
-  location: location
+  location: 'eastus'
```
Comment (AlexanderSehr) [inline avm/res/web/serverfarm/tests/e2e/max/dependencies.bicep:9]: why? Does it have to be in the same region as the server farm?

File: avm/res/web/serverfarm/README.md [hunk 3]
```diff
@@ -54,7 +54,7 @@ module serverfarm 'br/public:avm/res/web/serverfarm:<version>' = {
       tier: 'Premium'
     }
     // Non-required parameters
-    location: '<location>'
+    location: 'eastus'
```
Comment (AlexanderSehr) [inline avm/res/web/serverfarm/README.md:57]: Needs updating

File: avm/res/web/serverfarm/tests/e2e/max/dependencies.bicep (hunk 2 above)
Comment (ChrisSidebotham) [reply avm/res/web/serverfarm/tests/e2e/max/dependencies.bicep:9]: Surely keeping all of the resources together would be best for now so we can unblock the CI Env?

File: avm/res/web/serverfarm/README.md (hunk 3 above)
Comment (ChrisSidebotham) [reply avm/res/web/serverfarm/README.md:57]: This is generated from the test files what needs updating here?

Comment (AlexanderSehr) [reply avm/res/web/serverfarm/README.md:57]: Just testing you!

File: avm/res/web/serverfarm/tests/e2e/max/dependencies.bicep (hunk 2 above)
Comment (AlexanderSehr) [reply avm/res/web/serverfarm/tests/e2e/max/dependencies.bicep:9]: in theory yes - but if we start also changing the dependency files, cleaning this up later will become a lot harder (also harder to troubleshoot if we overlook a location).

Comment (AlexanderSehr) [reply avm/res/web/serverfarm/tests/e2e/max/dependencies.bicep:9]: In that case I'd rather suggest to hardcode it in the invocating deployment - i.e. in the `main.test.bicep`. Then the workaround is limited to one location

**Review by AlexanderSehr — CHANGES_REQUESTED** [review]

File: avm/res/web/serverfarm/tests/e2e/max/dependencies.bicep (hunk 2 above)
Comment (ChrisSidebotham) [reply avm/res/web/serverfarm/tests/e2e/max/dependencies.bicep]: updated

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**